lib/Option/SanitizerOptions.cpp

```diff
@@ -107,55 +117,65 @@ static bool isTSanSupported(
   return Triple.isArch64Bit() && sanitizerRuntimeLibExists("tsan");
 }
 
-SanitizerKind swift::parseSanitizerArgValues(const llvm::opt::Arg *A,
+OptionSet<SanitizerKind> swift::parseSanitizerArgValues(
+    const llvm::opt::ArgList &Args,
+    const llvm::opt::Arg *A,
     const llvm::Triple &Triple,
     DiagnosticEngine &Diags,
     llvm::function_ref<bool(llvm::StringRef)> sanitizerRuntimeLibExists) {
-  SanitizerKind kind = SanitizerKind::None;
+  OptionSet<SanitizerKind> sanitizerSet;
 
   // Find the sanitizer kind.
-  SanitizerKind pKind = SanitizerKind::None;
   for (int i = 0, n = A->getNumValues(); i != n; ++i) {
-    kind =
-    llvm::StringSwitch<SanitizerKind>(A->getValue(i))
-      .Case("address", SanitizerKind::Address)
-      .Case("thread", SanitizerKind::Thread)
-      .Default(SanitizerKind::None);
-
-    if (kind == SanitizerKind::None) {
+    std::string opt = A->getValue(i);
+    if (opt == "address") {
+      sanitizerSet |= SanitizerKind::Address;
+    } else if (opt == "thread") {
+      sanitizerSet |= SanitizerKind::Thread;
+    } else if (opt == "fuzzer") {
+      sanitizerSet |= SanitizerKind::Fuzzer;
+    } else {
       Diags.diagnose(SourceLoc(), diag::error_unsupported_option_argument,
-        A->getOption().getPrefixedName(), A->getValue(i));
-      return kind;
-    }
-
-    // Currently, more than one sanitizer cannot be enabled at the same time.
-    if (pKind != SanitizerKind::None && pKind != kind) {
-      SmallString<128> pb;
-      SmallString<128> b;
-      Diags.diagnose(SourceLoc(), diag::error_argument_not_allowed_with,
-        (A->getOption().getPrefixedName() + toStringRef(pKind)).toStringRef(pb),
-        (A->getOption().getPrefixedName() + toStringRef(kind)).toStringRef(b));
+          A->getOption().getPrefixedName(), A->getValue(i));
+      return sanitizerSet;
```

Comment:
Nitpick: there may still be valid sanitizers after this one…or more invalid ones that it would be good to warn about.